build/entrypoint.sh

```diff
@@ -16,4 +16,11 @@ if [ ! -f "$BROKER_CONFIG" ] ; then
 fi
 echo "Using config file mounted to $BROKER_CONFIG"
 
-asbd -c $BROKER_CONFIG $FLAGS
+cat << EOF  > "${REGISTRY_AUTH}"
+---
+- type: dockerhub
+  user: ${DOCKERHUB_USER}
+  pass: ${DOCKERHUB_PASS}
+EOF
+
+asbd -c $BROKER_CONFIG --registryauth=$REGISTRY_AUTH $FLAGS
```

Comment:
I prefer NOT to have this as a separate option. Why can't the admin specify the location of the secret in the registry config section?

Something like `auth: TYPE|NAME` where `TYPE` can be `secret` or `file`. Then `NAME` can be the name of the secret OR the fully qualified filename: `/etc/ansible-service-broker/dockerhub-auth`

```yaml
registry:
  - type: dockerhub
    name: docker
    url: https://registry.hub.docker.com
    auth: TYPE|NAME
    org: DOCKERHUB_ORG
    fail_on_error: false
  - type: rhcc
    name: rhcc
    url: registry.access.stage.redhat.com
    fail_on_error: true
    white_list:
      - "^legitimate.*-apb$"    
      # will allow all the APBs to be included. You must have at least 1 white 
      # list to retrieve APBs and this is the most permissive 
      - ".*-apb$" 
    black_list:
      - "malicious.*-apb$"
      - "^specific-blacklist-apb$"
dao:
```

OR we could use
```yaml
   auth_type: secret|file
   auth_name: NAME_OF_SECRET | FQFN
```

This gives us the flexibility to know how to read different ways. File makes it easier to run locally. But secret might make it easier in OpenShift by not needing a mount. Just go read the secret.

Reply:
OR we could reuse the auth format I did for basic auth and you might even be able to use the same structs (maybe not entire sure).

Reply:
+1, you're right no reason to make this a separate config. I agree with you. I think I pidgeonholed myself there because I was initially mounting the secret as a volume.